mainchain-follower/db-sync-follower/src/native_token/mod.rs

```diff
@@ -0,0 +1,58 @@
+use crate::db_model::Address;
+use crate::metrics::McFollowerMetrics;
+// use crate::observed_async_trait;
+use crate::db_model::SlotNumber;
+use async_trait::async_trait;
+use main_chain_follower_api::native_token::NativeTokenManagementReleaseEvent;
+use main_chain_follower_api::NativeTokenManagementDataSource;
+use main_chain_follower_api::Result;
+use sidechain_domain::*;
+use sqlx::PgPool;
+
+pub struct NativeTokenManagementDataSourceImpl {
+	pub pool: PgPool,
+	pub metrics_opt: Option<McFollowerMetrics>,
+}
+
+#[async_trait]
```

Comment:
Was there any particular problem with `observed_async_trait` ? If so, it's a good indicator to reimplement `observed_async_trait` without `async_trait` if possible.

Reply:
I dropped it, because it messed with my auto-completion, and didn't bring it back. I'll re-add it to verify it works

Reply:
Added it and it works